File: src/integration-tests/results/rs/complex.rs
extern crate serde_json;

pub type Skip = bool;
/// Bip
///
/// abc 123
///
/// # Default
///
/// 123
///
/// # Example
///
/// `123`
///
/// # Example
///
/// `456`
///
/// # Example
///
/// `789`
///
pub type Bip = i64;
pub type Bipper = f64;
pub type Bop = f64;
#[derive(Serialize, Deserialize)]
pub enum Bopper {
    #[serde(rename = asd)]
    Asd,
    #[serde(rename = bca)]
    Bca,
}
pub type Swicker = f64;
pub type Slicker = i64;
pub type Splinter = Vec<serde_json::Value>;
pub type Skibb = serde_json::Value;
pub type Skibbidippy = Vec<Bop>;
pub type Bopskippity = String;
#[derive(Serialize, Deserialize)]
pub struct Floopdidoop {
    pub(crate) fooberdoober: Bop,
    pub(crate) gibbledybits: Option<Bopskippity>,
}
#[derive(Serialize, Deserialize)]
pub struct Skiperydippery {
    pub(crate) fooberdoober: Option<Bop>,
    pub(crate) gibbledybits: Option<Bopskippity>,
}
pub type Doppler = HashMap<String, Option<serde_json::Value>>;
pub type Gorbelchov = f64;
#[derive(Serialize, Deserialize)]
pub struct Bonkiedonky {
    pub(crate) shopper: Option<Doppler>,
    pub(crate) badmirputin: Option<Gorbelchov>,
}
pub type Justworkalready = HashMap<String, Option<serde_json::Value>>;
pub type Zip = i64;
pub type Nullgasm = serde_json::Value;
#[derive(Serialize, Deserialize)]
pub enum Skorpionuts {
    Skip,
    Zip,
    Nullgasm
}
#[derive(Serialize, Deserialize)]
pub enum Chikypoops {
    Skip,
    Zip
}
#[derive(Serialize, Deserialize)]
pub struct BippyskippyBoppy {
    pub(crate) bool: Option<Skip>,
    pub(crate) int: Option<Bip>,
    pub(crate) number: Option<Bipper>,
    pub(crate) string: Option<Bop>,
    pub(crate) stringEnum: Option<Bopper>,
    pub(crate) numbericalEnum: Option<Swicker>,
    pub(crate) integerEnum: Option<Slicker>,
    pub(crate) untypedArray: Option<Splinter>,
    pub(crate) orderedArray: Option<Skibbidippy>,
    pub(crate) unorderedArray: Option<Skibbidippy>,
    pub(crate) object: Option<Floopdidoop>,
    pub(crate) allOf: Option<Justworkalready>,
    pub(crate) anyOf: Option<Skorpionuts>,
    pub(crate) oneOf: Option<Chikypoops>,
}

File: src/integration-tests/results/py/complex.py
from typing import NewType

Skip = NewType("Skip", bool)
from typing import NewType
"""abc 123
"""
Bip = NewType("Bip", int)
from typing import NewType

Bipper = NewType("Bipper", float)
from typing import NewType

Bop = NewType("Bop", float)
from enum import Enum

class Bopper(Enum):
    ASD = 0
    BCA = 1
from typing import NewType

Swicker = NewType("Swicker", float)
from typing import NewType

Slicker = NewType("Slicker", int)
from typing import List, Any, NewType

Splinter = NewType("Splinter", List[Any])
from typing import Any, NewType

Skibb = NewType("Skibb", Any)
from typing import List, NewType

Skibbidippy = NewType("Skibbidippy", List[Bop])
from typing import NewType

Bopskippity = NewType("Bopskippity", str)
from typing import TypedDict, Optional

class Floopdidoop(TypedDict):
    fooberdoober: Optional[Bop]
    gibbledybits: Optional[Bopskippity]
from typing import TypedDict, Optional

class Skiperydippery(TypedDict):
    fooberdoober: Optional[Bop]
    gibbledybits: Optional[Bopskippity]
from typing import NewType, Any, Mapping

Doppler = NewType("Doppler", Mapping[Any, Any])
from typing import NewType

Gorbelchov = NewType("Gorbelchov", float)
from typing import TypedDict, Optional

class Bonkiedonky(TypedDict):
    shopper: Optional[Doppler]
    badmirputin: Optional[Gorbelchov]
from typing import NewType, Any, Mapping

Justworkalready = NewType("Justworkalready", Mapping[Any, Any])
from typing import NewType

Zip = NewType("Zip", int)
from typing import NewType

Nullgasm = NewType("Nullgasm", None)
from typing import NewType, Union

Skorpionuts = NewType("Skorpionuts", Union[Skip, Zip, Nullgasm])
from typing import NewType, Union

Chikypoops = NewType("Chikypoops", Union[Skip, Zip])
from typing import TypedDict, Optional

class BippyskippyBoppy(TypedDict):
    bool: Optional[Skip]
    int: Optional[Bip]
    number: Optional[Bipper]
    string: Optional[Bop]
    stringEnum: Optional[Bopper]
    numbericalEnum: Optional[Swicker]
    integerEnum: Optional[Slicker]
    untypedArray: Optional[Splinter]
    orderedArray: Optional[Skibbidippy]
    unorderedArray: Optional[Skibbidippy]
    object: Optional[Floopdidoop]
    allOf: Optional[Justworkalready]
    anyOf: Optional[Skorpionuts]
    oneOf: Optional[Chikypoops]
